proposed/event-dispatcher-meta.md

```diff
@@ -37,85 +37,86 @@ The Working Group identified four possible workflows for event passing, based on
+* Indicating that some change in system configuration or some user action has occurred and allowing other systems to react in ways that do not affect program flow (such as sending an email or logging the action).
+* Passing an object to a series of Listeners to allow it to be modified before it is saved to a persistence system.
+* Passing a collection to a series of Listeners to allow them to register values with it so that the Emitter may act on all of the collected information.
+* Passing a collection to a series of Listeners to allow them to modify the collection in some way before the Emitter takes action.
+* Passing some contextual information to a series of Listeners so that all of them may "vote" on what action to take, with the Emitter deciding based on the aggregate information provided.
+* Passing an object to a series of listeners and allowing any listener to terminate the process early before other listeners have completed.
 
 ### 4.3 Immutable events
 
-Initially the Working Group wished to define all Events as immutable message objects, similar to PSR-7.  However, that proved problematic in the Processor case.  In that case Listeners needed a way to return data to the caller.  In concept there were three possible avenues:
+Initially the Working Group wished to define all Events as immutable message objects, similar to PSR-7.  However, that proved problematic in all but the one-way notification case.  In the other scenarios Listeners needed a way to return data to the caller.  In concept there were three possible avenues:
 
 * Make the event mutable and modify it in place.
 * Require that events be evolvable (immutable but with `with*()` methods like PSR-7 and PSR-13) and that listeners return the event to pass along.
 * Make the Event immutable but aggregate and return the return value from each Listener.
 
-However, Stoppable Task (the alternative chain case) also needed to have a channel by which to indicate that further listeners should not be called.  That could be done either by:
+However, Stoppable Events (the alternative chain case) also needed to have a channel by which to indicate that further listeners should not be called.  That could be done either by:
 
-* Modifying the Task (`stopPropagation()`)
+* Modifying the Event (e.g., calling a `stopPropagation()` method)
 * Returning a sentinel value from the listener (`true` or `false`) to indicate that propagation should terminate.
-* Evolving the Task to be stopped (`withPropagationStopped()`)
+* Evolving the Event to be stopped (`withPropagationStopped()`)
+
+Of those alternatives, therefore, the first would mandate a mutable Event in at least some cases.  The second would mandate a mutable Event as the return value was already in use.  And the third seemed unnecessarily ceremonial and pedantic as it would entail additional syntax requirements for developers with little added value.
```

Comment:
Why does the second mandate a mutable event? And how is the return value already in use - we currently _ignore_ return values from listeners (which is where the sentinel value would be returned from)?

I suggest the following for the above paragraph:

> Each of these alternatives have drawbacks. The first means that, at least for the purposes of indicating propagation status, events must be mutable. The second requires that listeners return a value, at least when they intend to halt event propagation; this could have ramifications with existing libraries, and potential issues in terms of documentation. The third requires that listeners return the event or mutated event in _all_ cases, and would require dispatchers to test to ensure that the returned value is of the same type as the value passed to the listener; it effectively puts an onus both on consumers and implementers, and thus raising more potential integration issues.